### PR TITLE
Feature/url parsing

### PR DESCRIPTION
**Pull request by Tiriel**
Please feel free to suggest changes!

From #22 :
- [x] Add `yargs`
- [x] General name and purpose of the command
- [x] Basic usage example
- [x] Arguments details

From #25 :
- [x] When first arg is url, extract owner, repo, and pr-id from it
- When first arg is integer:
  - [x] Check second arg for owner and third arg for repo if given
  - [x] Default owner to `nodejs` and repo to `node` if not
  - [x] Allow args to be given in any other order when named? (`--repo=node --owner=nodejs`)

I have tried to test it manually, seems to be working as expected!

### CODE REVIEW

File: bin/metadata.js [hunk 1]
```diff
@@ -3,36 +3,57 @@
 
 const fs = require('fs');
 const path = require('path');
+const argv = require('yargs').
```
Comment (joyeecheung) [inline bin/metadata.js]: Ah..just wondered why travis passed with this style, turns out we do not run `npm run lint` in travis.yml...can you run `npm run lint` and fix the styles? Thanks!

**Comment by Tiriel** [comment]
Fixed lint! Sorry, don't know why my IDE linted it that way...

**Comment by joyeecheung** [comment]
I have given it a try and looks like it's not working as intended...

screen shot 2017-10-31 at 10 44 48 pm
screen shot 2017-10-31 at 10 43 50 pm

File: bin/metadata.js [hunk 2]
```diff
@@ -3,36 +3,57 @@
...
 const MetadataGenerator = require('../lib/metadata_gen');
 
 // const REFERENCE_RE = /referenced this pull request in/
 
-const PR_ID = parsePRId(process.argv[2]);
-const OWNER = process.argv[3] || 'nodejs';
-const REPO = process.argv[4] || 'node';
+let OWNER;
+let REPO;
+const PR_ID = parsePRId(argv._[0]);
```
Comment (joyeecheung) [inline bin/metadata.js]: This can be just `argv.identifier`

File: bin/metadata.js [hunk 3]
```diff
@@ -86,8 +107,16 @@ main(PR_ID, OWNER, REPO).catch((err) => {
 
 function parsePRId(id) {
   // Fast path: numeric string
-  if (`${+id}` === id) { return +id; }
-  const match = id.match(/^https:.*\/pull\/([0-9]+)(?:\/(?:files)?)?$/);
-  if (match !== null) { return +match[1]; }
+  if (!isNaN(id)) {
+    OWNER = argv._[1] || argv.o || 'nodejs';
```
Comment (joyeecheung) [inline bin/metadata.js]: We need to test if `options` is a string, if it is, then it's `owner`

File: bin/metadata.js [hunk 4]
```diff
@@ -86,8 +107,16 @@ main(PR_ID, OWNER, REPO).catch((err) => {
...
+    REPO = argv._[2] || argv.r || 'node';
```
Comment (joyeecheung) [inline bin/metadata.js]: This would be in `argv._[0]` actually (if supplied)

**Comment by joyeecheung** [comment]
Also..we don't have a test for CLI, that's why this passes travis. I think we should get some tests in before this lands.

File: bin/metadata.js (hunk 2 above)
Comment (Tiriel) [reply bin/metadata.js]: Yup, got it at the same time, I changed the yargs.usage last minute, my bad

File: bin/metadata.js (hunk 3 above)
Comment (Tiriel) [reply bin/metadata.js]: I directly changed usage to be more verbose and add the distinction between owner and repo

File: bin/metadata.js (hunk 4 above)
Comment (Tiriel) [reply bin/metadata.js]: Like previous comment!

**Comment by Tiriel** [comment]
True that.

This brings a question. I have no idea what's usually used for testing features in node "core" and such. I'm used to mocha but I guess we're note using it, to avoid adding more deps. How should I do?

**Comment by Tiriel** [comment]
Okay. Lint ok, tests ok, build ok... At last.

Sorry for the messy work, not used to this sort of thing yet, learning.
Good for review I guess!

**Comment by joyeecheung** [comment]
@Tiriel I have merged the refactoring PR, there is a conflict now, can you rebase?

**Comment by joyeecheung** [comment]
BTW this is how you can rebase (https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#step-5-rebase)

```
$ git checkout feature/url-parsing
$ git fetch upstream # assuming upstream points to this repo
$ git rebase upstream/master
# fix conflicts
$ git push --force origin feature/url-parsing  # assuming origin points to your fork
```

**Comment by Tiriel** [comment]
No problem, doing it right now! And thanks for the tip.

**Comment by Tiriel** [comment]
Done!

Linted, tests passing on my local, command working as far as I can tell!

**Comment by joyeecheung** [comment]
Awesome work! Thanks!

**Pull request merged**